presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -745,6 +746,61 @@ public void testViewMetadata()
         assertUpdate("DROP VIEW meta_test_view");
     }
 
+    @Test
+    public void testCurrentUserInView()
+    {
+        skipTestUnless(supportsViews());
+        if (!getSession().getCatalog().isPresent() || !getSession().getSchema().isPresent()) {
+            return;
+        }
+        String testAccountsUnqualifiedName = "test_accounts";
+        String testAccountsViewUnqualifiedName = "test_accounts_view";
+        // String testAccountsFullyQualifiedName = format("%s.%s.%s", getSession().getCatalog().get(), getSession().getSchema().get(), testAccountsUnqualifiedName);
+        String testAccountsViewFullyQualifiedName = format("%s.%s.%s", getSession().getCatalog().get(), getSession().getSchema().get(), testAccountsViewUnqualifiedName);
+        assertUpdate(format("CREATE TABLE %s AS SELECT user_name, account_name" +
+                "  FROM (VALUES ('user1', 'account1'), ('user2', 'account2'))" +
+                "  t(user_name, account_name)", testAccountsUnqualifiedName), 2);
+        assertUpdate(format("CREATE VIEW %s AS SELECT account_name FROM test_accounts WHERE user_name = CURRENT_USER", testAccountsViewUnqualifiedName));
+        try {
+            assertUpdate(format("GRANT SELECT ON %s TO user1", testAccountsViewFullyQualifiedName));
+        }
+        catch (Exception e) {
+            try {
+                assertUpdate(format("GRANT SELECT ON %s TO user1", testAccountsViewUnqualifiedName));
+            }
+            catch (Exception e2) {
+                // ignored - some connectors do not create the table in the same schema as the session
```

Comment:
This seems broken. What connectors do that?

Reply:
The hive connector fails without the fully qualified name and raptor fails with the fully qualified name.